src/Catalyst.Core.Modules.Dfs/WebApi/V0/Controllers/FileSystemController.cs

```diff
@@ -0,0 +1,294 @@
+
+        /// <summary>
+        ///   Get the object as a TAR file.
+        /// </summary>
+        /// <param name="arg">
+        ///   A path to an existing file or directory.
+        /// </param>
+        /// <param name="compress">
+        ///   If <b>true</b>, generate gzipped TAR.
+        /// </param>
+        [HttpGet, HttpPost, Route("get")]
+        [Produces("application/tar")]
+        public async Task Get(string arg, bool compress = false)
+        {
+            var tar = await IpfsCore.UnixFsApi.GetAsync(arg, compress, Cancel);
+            Response.ContentType = "application/tar";
+            Response.Headers.Add("X-Stream-Output", "1");
+            Response.Headers.Add("X-Content-Length", "4");
+            Response.StatusCode = 200;
+
+            await tar.CopyToAsync(Response.Body);
+            await Response.Body.FlushAsync();
+        }
+
+        /// <summary>
+        ///   Get information on the file or directory.
+        /// </summary>
+        /// <param name="arg">
+        ///   A path to an existing file, such as "QmXarR6rgkQ2fDSHjSY5nM2kuCXKYGViky5nohtwgF65Ec/about"
+        ///   or "QmZTR5bcpQD7cFgTorqxZDYaew1Wqgfbd2ud9QqGPAkK2V"
+        /// </param>
+        [HttpGet, HttpPost, Route("file/ls")]
+        public async Task<FileSystemDetailsDto> Stat(string arg)
+        {
+            var node = await IpfsCore.UnixFsApi.ListFileAsync(arg, Cancel);
+            var dto = new FileSystemDetailsDto
+            {
+                Arguments = new Dictionary<string, string>(),
+                Objects = new Dictionary<string, FileSystemDetailDto>()
+            };
+            dto.Arguments[arg] = node.Id;
+            dto.Objects[node.Id] = new FileSystemDetailDto
+            {
+                Hash = node.Id,
+                Size = node.Size,
+                Type = node.IsDirectory ? "Directory" : "File",
+                Links = node.Links
+                   .Select(link => new FileSystemLinkDto
+                    {
+                        Hash = link.Id,
+                        Name = link.Name,
+                        Size = link.Size
+                    })
+                   .ToArray()
+            };
+            return dto;
+        }
+
+        /// <summary>
+        ///   Add a file.
+        /// </summary>
+        [HttpGet, HttpPost, Route("add")]
+        public async Task Add(IFormFile file,
+            string hash = MultiHash.DefaultAlgorithmName,
+            [ModelBinder(Name = "cid-base")] string cidBase = MultiBase.DefaultAlgorithmName,
+            [ModelBinder(Name = "only-hash")] bool onlyHash = false,
+            string chunker = null,
+            bool pin = false,
+            [ModelBinder(Name = "raw-leaves")] bool rawLeaves = false,
+            bool trickle = false,
+            [ModelBinder(Name = "wrap-with-directory")]
+            bool wrap = false,
+            string protect = null,
+            bool progress = true)
+        {
+            if (file == null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)